### PR TITLE
feat(doc-core): doc support nav postion(#3604)

### PR DESCRIPTION
**Pull request by RyanHo97**
## Summary

doc support nav postion(#3604)

## Details

Add a position param in nav config

## Related Issue

Issue number: #3604

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/doc-tools-doc/modern.config.ts [hunk 1]
```diff
@@ -140,11 +140,13 @@ function getNavbar(): NavItem[] {
       text: getI18nKey('guide'),
       link: '/guide/getting-started',
       activeMatch: '/guide/',
```
Comment (sanyuan0704) [inline packages/document/doc-tools-doc/modern.config.ts:142]: I think you can restore the doc config for debug.

**Review by sanyuan0704 — APPROVED** [review]

File: packages/document/doc-tools-doc/modern.config.ts (hunk 1 above)
Comment (RyanHo97) [reply packages/document/doc-tools-doc/modern.config.ts:142]: Thank you. I have resubmitted the code and it seems that the submitted code still has this issue：
Some checks were not successful

File: packages/document/doc-tools-doc/docs/zh/guide/navbar.mdx [hunk 2]
```diff
@@ -42,24 +43,38 @@ export interface NavItemWithLink {
...
   items: (NavItemWithLink | NavItemWithChildren)[];
+  position?: 'left' | 'right';
 }
 ```
 
 其中各项属性的含义如下:
 
 - `text` - 导航栏文本
 - `items` - 子导航栏元素
+- `position`-带有子元素的导航栏菜单项定位
```
Comment (sanyuan0704) [inline packages/document/doc-tools-doc/docs/zh/guide/navbar.mdx:77]: There is a space after `-`

File: packages/document/doc-tools-doc/docs/en/guide/navbar.mdx [hunk 3]
```diff
@@ -34,29 +34,47 @@ export interface NavItemWithLink {
...
   items: NavItem[];
+  position?: 'left' | 'right';
 }
 ```
 
 The meanings of the attributes are as follows:
 
--`text` - Navbar text -`items` - subnavbar elements
+- `text` - Navbar text
+- `items` - subnavbar elements
```
Comment (sanyuan0704) [inline packages/document/doc-tools-doc/docs/en/guide/navbar.mdx:76]: Subnavbar

**Review by sanyuan0704 — APPROVED** [review]

**Pull request merged**